File: APP/TaQuot/src/models/prestation/prestation.ts
export default class Prestation {
  id = "";
  issueGitLab = 0;
  displayNumber = 0;
  commentaire = "";
  tache = "";
  // date = '' ;
  date = new Date();
  temps = 0;
  typeTache = "";
  commanditaire = "";
  application = "";
  module = "";

  get formatedDate(): string {
    if (this.date == null || this.date == undefined) return ``;

    return `${this.date.getDate()}/${
      this.date.getMonth() + 1
    }/${this.date.getFullYear()}`;
  }
}

File: API/AcQua TaQuot/Entities/Models/Prestation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models
{
    public class Prestation
    {
        [Column("PrestationId")]
        public Guid Id { get; set; }
        public int IssueGitLab { get; set; }
        public int? DisplayNumber { get; set; }
        public string? Commentaire { get; set; }
        [Required()]
        public DateTime Date { get; set; }
        [Required()]
        public int Temps { get; set; }
        public bool IsFacturable { get; set; }
        [MaxLength(3, ErrorMessage = "Trigramme utilisateur uniquement")]
        [MinLength(3, ErrorMessage = "Trigramme utilisateur uniquement")]
        [Required()]
        public string Utilisateur { get; set; }

        [ForeignKey(nameof(TypeTache))]
        public Guid TypeTacheId { get; set; }
        public TypeTache? TypeTache { get; set; }

        [ForeignKey(nameof(Statut))]
        public Guid StatutId { get; set; }
        public Statut? Statut { get; set; }

        public Tache? Tache { get; set; }
        public DateTime? DateFacturation { get; set; } = null;
        public ICollection<LigneFacture>? LignesFacture { get; set; }
    }
}
